src/Features/CSharp/Portable/ConvertAnonymousTypeToTuple/CSharpConvertAnonymousTypeToTupleDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.CSharp.ConvertAnonymousTypeToTuple
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal class CSharpConvertAnonymousTypeToTupleCodeRefactoringProvider : AbstractCodeStyleDiagnosticAnalyzer
```

Comment:
the name seems rather fishy

Reply:
good catch. i flip-flopped on this being an analyzer/refactoring and didn't fix up the name.